File: we_12_1_projecting_a_point_with_the_pinhole.py
# Integrated Navigation Systems — Chapter 12: Visual, LiDAR, and Radar-Inertial Odometry and SLAM
# Worked Example 12.1 — Projecting a point with the pinhole model
# Companion code (educational reference implementation; see the book for context).

fx,cx,X,Z=500,320,0.5,2.0
print(f'u={fx*(X/Z)+cx:.0f} px')   # 445

File: we_12_1_projecting_a_point_with_the_pinhole.m
% Integrated Navigation Systems — Chapter 12: Visual, LiDAR, and Radar-Inertial Odometry and SLAM
% Worked Example 12.1 — Projecting a point with the pinhole model
% Companion code (educational reference implementation; see the book for context).

fx=500; cx=320; X=0.5; Z=2.0;
u=fx*(X/Z)+cx; fprintf('u=%.0f px\n', u);   % 445
